es/development/debugging.rst

```diff
@@ -1,16 +1,210 @@
-Debugging
-#########
+Depuración
+##########
 
-.. note::
-    La documentación no es compatible actualmente con el idioma español en esta página.
+La depuración es una parte inevitable y necesaria de cualquier ciclo de desarrollo.
+Aunque CakePHP no ofrece ninguna herramienta que se conecte directamente
+con algun IDE o editor, CakePHP proporciona varias herramientas para
+asistirte en la depuración y exponer lo que se está ejecutando bajo el capó de
+tu aplicación.
 
-    Por favor, siéntase libre de enviarnos un pull request en
-    `Github <https://github.com/cakephp/docs>`_ o utilizar el botón **Improve this Doc** para proponer directamente los cambios.
+Depuración Básica
+=================
 
-    Usted puede hacer referencia a la versión en Inglés en el menú de selección superior
-    para obtener información sobre el tema de esta página.
+.. php:function:: debug(mixed $var, boolean $showHtml = null, $showFrom = true)
+    :noindex:
+
+La función ``debug()`` es una función que está disponible globalmente y funciona
+de manera similar a la función ``print_r()`` de PHP. La función ``debug()``
+te permite mostrar el contenido de una variable de varias maneras.
+Primero, si deseas que los datos se muestren de una forma amigable con HTML, 
+debes establecer el segundo parámetro en ``true``. La función
+también imprime la línea y el archivo de origen por defecto.
+
+El resultado de esta función solo se mostrará si la variable ``$debug`` en el archivo core es ``true``.
+
+.. versionadded:: 3.3.0
+
+    Al llamar a este método se devolverá ``$var`` que se le haya pasado, para que pueda, por ejemplo,
+    colocarlo en declaraciones de retorno, por ejemplo:
```

Comment:
You need to end this paragraph with `::` to make the next line a formatted code block.

Reply:
Solved